model/fv_dynamics.F90

```diff
@@ -747,8 +768,68 @@ subroutine fv_dynamics(npx, npy, npz, nq_tot,  ng, bdt, consv_te, fill,
                      inline_mp, flagstruct%c2l_ord, bd, flagstruct%fv_debug, &
                      flagstruct%moist_phys)
 
+#ifdef MOLECULAR_DIFFUSION
+! do thermosphere adjustment if it is turned on and at last_step.
```

Comment:
Same as @lharris4 comment above regarding turning this block of code into a subroutine within the newly created molecular diffusion source file and not inlined here.